ui/imports/shared/stores/PermissionsStore.qml

```diff
@@ -14,16 +15,48 @@ QtObject {
     readonly property var permissionsModel:
         chatCommunitySectionModuleInst.permissionsModel
 
-    readonly property var becomeMemberPermissionsModel: SortFilterProxyModel {
-        id: becomeMemberPermissionsModel
+    readonly property bool viewOnlyCanAddReaction: (qSettings.viewOnlyUserAllowedAddReaction && root.tokenCriteriaMet)
+
+    property var qSettings: Settings {
+        property bool viewOnlyUserAllowedAddReaction: false
+    }
+
+    function setPostRections(checked) {
+        qSettings.viewOnlyUserAllowedAddReaction = checked;
+    }
+
+    readonly property bool viewAndPostCriteriaMet: {
+        for (var i = 0; i < selectedChannelPermissionsModel.count; i++) {
+            var permissionItem = selectedChannelPermissionsModel.itemAt(i);
+            if (permissionItem && permissionItem.tokenCriteriaMet)
+                return true
+        }
+        return false
+    }
+
+    readonly property bool tokenCriteriaMet: {
+        for (var i = 0; i < selectedChannelPermissionsModel.count; i++) {
+            var permissionItem = selectedChannelPermissionsModel.itemAt(i);
+            if (permissionItem && permissionItem.tokenCriteriaMet)
+                return true
+        }
+        return false
+    }
```

Comment:
How are these 2 properties `viewAndPostCriteriaMet` and `tokenCriteriaMet` different when they contain the same code?